subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyPermissionsIntegrationTest.groovy

```diff
@@ -222,4 +223,34 @@ class CopyPermissionsIntegrationTest extends AbstractIntegrationSpec {
         cleanup:
         file('src/unauthorized').mode = 0777
     }
+
+    @Requires(TestPrecondition.FILE_PERMISSIONS)
```

Comment:
Do we have to restrict this to *NIX? I'd expect Windows be able to create a file that's not readable. Maybe you need to call `File.setReadable(false, false)` for that?

Reply:
You mean change the definition of `TestPrecondition.FILE_PERMISSIONS` to include Windows?
Or you mean that we don't actually need file permissions for this test to work?

Reply:
Added back, was failing on Windows: https://builds.gradle.org/viewLog.html?buildId=24326586&buildTypeId=Gradle_Check_Stage_ReadyforMerge_Trigger

Reply:
Have you tried using `File.setReadable(false, false)` on Windows?

Reply:
It can't revoke the rights, apparently.
I've changed it to `setReadable(false, false)` anyway and added an assertion for readability, as explanation.
<img width="792" alt="Screen Shot 2019-06-28 at 15 26 51" src="https://user-images.githubusercontent.com/1841944/60342075-2cbb9080-99b9-11e9-993b-b992802c44ce.png">

Reply:
:+1: thanks.